server/src/domain/library/library.service.ts

```diff
@@ -590,106 +619,96 @@ export class LibraryService extends EventEmitter {
   async handleQueueAssetRefresh(job: ILibraryRefreshJob): Promise<JobStatus> {
```

Comment:
This function is currently hard to read. Maybe you could split it up into smaller functions to make it easier to read

Reply:
I do agree that this function is growing more and more complex, but this PR does not greatly increase that complexity IMO. Perhaps I can refactor the function further in a separate PR?